pkg/codegen/go/gen.go

```diff
@@ -1477,6 +1482,14 @@ func parseEnvFloat(v string) interface{} {
 	return f
 }
 
+func parseEnvStringArray(v string) interface{} {
+	var result []pulumi.StringInput
+	for _, item := range strings.Split(v, ";") {
+		result = append(result, pulumi.StringInput(pulumi.String(item)))
```

Comment:
`pulumi.String` implements `pulumi.StringInput`: https://github.com/pulumi/pulumi/blob/00f8966a9a5e8b693cf41c1e78457956a8b8ec33/sdk/go/pulumi/types_builtins.go#L6515

You should be able to simplify all of this:

```go
var result pulumi.StringArray
...
result = append(result, pulumi.String(item))
```

Several different incantations of this will work FWIW.

Reply:
That's a good one. I tried a couple of combos but I didn't try returning a `StringArray` directly.